src/main/java/eu/europeana/metis/sandbox/repository/DatasetRepository.java

```diff
@@ -1,8 +1,19 @@
 package eu.europeana.metis.sandbox.repository;
 
 import eu.europeana.metis.sandbox.entity.DatasetEntity;
+import eu.europeana.metis.sandbox.entity.projection.DatasetIdView;
+import java.time.LocalDateTime;
+import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface DatasetRepository extends JpaRepository<DatasetEntity, Integer> {
 
+  /**
+   * Get a list of datasets created before specified date
+   *
+   * @param date must not be null
+   * @return list of dataset ids
+   * @see DatasetIdView
+   */
+  List<DatasetIdView> getByCreatedDateBefore(LocalDateTime date);
```

Comment:
Just curiosity: where do you define the query that does this operation?

Reply:
I don't, [Spring Data](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation) defines it

Reply:
I see. Both cool and scary... :)
Could you perhaps put that link in the comments there, so that we can find this?

Reply:
Yes I will. 
Also you might notice that for other methods in other repositories with `deleteByDatasetId` I defined the query with the @Query annotation, the reason for that is that if you let Spring Data to define that one for you it will first perform a select of the records based on that property, and then delete one by one (because when doing a delete a list of deleted records is returned), and that takes more time than just what I wanted which was `delete from table where datasetId =` but in trade only the number of deleted records can be returned, which I do not use anyways so you see my returns are void.
So in my experience when using this feature, test locally changing `logback-spring.xml` with this 
``` <logger name="org.hibernate" level="DEBUG"/>```
So you can verify how the generated query will look like. And if you find out is not optimal then you can define your own using the @Query annotation